tests/error/poly_errors/non_linear2.err

```diff
@@ -3,5 +3,5 @@ Guppy compilation failed. Error in file $FILE:23
 21:    @guppy(module)
 22:    def main() -> None:
 23:        foo(h)
-               ^
-GuppyTypeError: Expected argument of type `?T -> ?T`, got `qubit @owned -> qubit`
+           ^^^^^^
+GuppyTypeError: Cannot instantiate non-linear type variable `T` in type `forall T. (T -> T) -> None` with linear type `qubit`
```

Comment:
A side effect of the unification change is that we get a better error message here.

However, I think #487 still applies